sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusProcessor.cs

```diff
@@ -700,6 +694,16 @@ private void ReconcileReceiverManagers(int maxConcurrentSessions)
                         }
                     }
                 }
+
+                int receiverManagers = _receiverManagers.Count;
+                for (int i = 0; i < receiverManagers; i++)
+                {
+                    var receiverManager = _receiverManagers[i];
+                    if (receiverManager.Receiver != null && receiverManager.Receiver.PrefetchCount != prefetchCount)
```

Comment:
This check is a bit weird because the SessionReceiveManager only associates the receiver at a later stage. This is not a problem though because the session manager has already a reference to the adjusted processor options and will take it into account once opened.